config/config.go

```diff
@@ -8,56 +8,100 @@ import (
 	"errors"
 	"flag"
 	"fmt"
-	"io/ioutil"
+	"os"
 
 	"github.com/elastic/elastic-agent-client/v7/pkg/client"
 	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
 	"github.com/elastic/elastic-agent-libs/config"
 	"github.com/elastic/elastic-agent-libs/logp"
-	"github.com/elastic/go-ucfg/json"
 
 	"github.com/elastic/elastic-agent-shipper/monitoring"
 	"github.com/elastic/elastic-agent-shipper/output"
 	"github.com/elastic/elastic-agent-shipper/queue"
-	"github.com/elastic/elastic-agent-shipper/server"
+
+	"go.uber.org/zap/zapcore"
 )
 
 var (
-	configFilePath    string
+	configFilePath string
+	// ErrConfigIsNotSet reports that no unmanaged config file has been set
 	ErrConfigIsNotSet = errors.New("config file is not set")
+	fs                = flag.CommandLine
+	// Overwrites is the config map of CLI overwrites set by the -E flag
+	Overwrites = config.SettingFlag(fs, "E", "config overwrites")
+
+	esKey      = "elasticsearch"
+	consoleKey = "console"
+	kafaKey    = "kafka"
 )
 
 // A lot of the code here is the same as what's in elastic-agent, but it lives in an internal/ library
 func init() {
-	fs := flag.CommandLine
 	fs.StringVar(&configFilePath, "c", "", "Run the shipper in the unmanaged mode and use the given configuration file instead")
+
+}
+
+// ShipperClientConfig is the shipper-relevant portion of the config received from input units
+type ShipperClientConfig struct {
+	Server string     `config:"server" mapstructure:"server"`
```

Comment:
Do we need both the `config` and `mapstructure` tags? Why not just one?

Reply:
I see we aren't using `mapstructure` everywhere either.

Reply:
I'm not sure that having two separate dependencies for config parsing makes this more maintainable.

There should be one way to parse config, even if that way is just the `config` package we don't like as much.

Reply:
Ah, yah, I tried to swap over to mapstructure, but ran into a few issues because the imported ES client code uses so many `config` struct tags. Can revert that.

Reply:
Considering that the only blocker for this is the upstream ES configs that live in beats, I wonder if we should just update those and then swap over to `mapstructure`....

Reply:
What needs to change in the ES configs? We can't make breaking changes to them, and we also don't want to fix mixing packages here by just doing it in Beats instead.

Reply:
Ah, just realized this was probably before Fae refactored the output, looks like the only component we're still importing from beats is the Kerberos config at `github.com/elastic/beats/v7/libbeat/common/transport/kerberos`.

Reply:
Alright, just dropped this until the fateful day we decide to move away from the config library.